scrapy_statsd_extension/__init__.py

```diff
@@ -0,0 +1,41 @@
+import statsd
+from scrapy import signals
+from twisted.internet import task
+
+
+class StatsdExtension(object):
+
+    def __init__(self, crawler):
+        host = crawler.settings.get('STATSD_HOST', 'localhost')
+        port = crawler.settings.get('STATSD_PORT', 8125)
+
+        self.client = statsd.StatsClient(host, port)
+        self.stats = crawler.stats
+        self._last_stats = {}
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        ext = cls(crawler)
+
+        crawler.signals.connect(
+            ext.spider_opened, signal=signals.spider_opened)
+        crawler.signals.connect(
+            ext.spider_closed, signal=signals.spider_closed)
+
+        return ext
+
+    def spider_opened(self, spider):
+        self.task = task.LoopingCall(self.log_stats, spider)
+        self.task.start(10)
+
+    def log_stats(self, spider):
+        for key, value in self.stats.get_stats().items():
+            if isinstance(value, int) or isinstance(value, float):
+                increment_amount = value - self._last_stats.get(key, 0)
+                self.client.incr(key, increment_amount)
```

Comment:
What are your thoughts on using counter stats vs gauges?

Reply:
For most metrics I think counters are what we need, a gauge will remain at the last value it's set until changed which can cause problems:
A workaround might be to set all back to 0 at the end of a job but this can cause lots of other headaches if the job finishes abruptly so I think it's better to stick with increments

Reply:
Okay, that's a good point.  I was thinking keeping track of _last_stats is kind of messy and most monitoring tools (ex: grafana) have a "diff" method which will take gauges and convert them to increments, but I forgot that statsd keeps on writing gauge values forever.